cmp/grid/Grid.js

```diff
@@ -202,7 +206,9 @@ class GridLocalModel extends HoistModel {
                 agColumnHeader: (props) => columnHeader({gridModel: model, ...props}),
                 agColumnGroupHeader: (props) => columnGroupHeader(props)
             },
-            rowSelection: model.selModel.mode,
+            rowSelection: selModel.mode,
+            suppressRowClickSelection: !selModel.isEnabled,
+            isRowSelectable: () => selModel.isEnabled,
```

Comment:
This could be an interesting closure/concept to bake into the selModel, although I suppose the current pattern would be to react to selection and filter/reset to reject any records you don't want to be selectable.  

Perhaps that's all we want and is the most appropriate/reactive approach.